### PR TITLE
cdc: add description about canal-json content-compatible.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

TiCDC introduces content-compatible for Canal-JSON. 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

https://github.com/pingcap/tiflow/pull/10014

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by 3AceShowHand** [comment]
该功能已经在 master，未进入 7.5.0 版本。

cherry-pick 回去 6.5.6， 7.1.3, 7.5.1

**Review by nongfushanquan — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

File: ticdc/ticdc-canal-json.md [hunk 1]
```diff
@@ -302,6 +302,13 @@ TiCDC 对 Canal-JSON 数据格式的实现，包括 `Update` 类型事件和 `my
 | `Update` 类型事件 | `old` 字段默认包含所有列的数据。当 sink 参数 `only_output_updated_columns` 设置为 `true` 时，`old` 字段仅包含被修改的列数据 | `old` 字段仅包含被修改的列数据          |
 | `mysqlType` 字段  | 对于含有参数的类型，没有类型参数信息                                                           | 对于含有参数的类型，会包含完整的参数信息 |
 
+### 兼容官方实现
```
Comment (qiancai) [inline ticdc/ticdc-canal-json.md:305]:
```suggestion
### 兼容 Canal 官方实现
```

File: ticdc/ticdc-canal-json.md [hunk 2]
```diff
@@ -302,6 +302,13 @@ TiCDC 对 Canal-JSON 数据格式的实现，包括 `Update` 类型事件和 `my
...
+
+自 v6.5.6 开始，TiCDC Canal-JSON 支持兼容官方输出的内容格式。在创建 changefeed 时，你可以在 `sink-uri` 中设置 `content-compatible=true` 以开启兼容模式。在该模式下，TiCDC 输出兼容官方实现的 Canal-JSON 格式数据。具体改动包括：
```
Comment (qiancai) [inline ticdc/ticdc-canal-json.md:307]:
```suggestion
自 v6.5.6 开始，TiCDC Canal-JSON 支持兼容 Canal 官方输出的内容格式。在创建 changefeed 时，你可以在 `sink-uri` 中设置 `content-compatible=true` 以开启兼容模式。在该模式下，TiCDC 输出兼容官方实现的 Canal-JSON 格式数据。具体改动包括：
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**